libraries/networking/src/udt/CongestionControl.cpp

```diff
@@ -0,0 +1,205 @@
+//
+//  CongestionControl.cpp
+//  libraries/networking/src/udt
+//
+//  Created by Clement on 7/23/15.
+//  Copyright 2015 High Fidelity, Inc.
+//
+//  Distributed under the Apache License, Version 2.0.
+//  See the accompanying file LICENSE or http://www.apache.org/licenses/LICENSE-2.0.html
+//
+
+#include "CongestionControl.h"
+#include "Packet.h"
+
+using namespace udt;
+using namespace std::chrono;
+
+static const double USECS_PER_SECOND = 1000000.0;
+
+void CongestionControl::setPacketSendPeriod(double newSendPeriod) {
+    Q_ASSERT_X(newSendPeriod >= 0, "CongestionControl::setPacketPeriod", "Can not set a negative packet send period");
+    
+    if (_maxBandwidth > 0) {
+        // anytime the packet send period is about to be increased, make sure it stays below the minimum period,
+        // calculated based on the maximum desired bandwidth
+        double minPacketSendPeriod = USECS_PER_SECOND / (((double) _maxBandwidth) / _mss);
+        _packetSendPeriod = std::max(newSendPeriod, minPacketSendPeriod);
+    } else {
+        _packetSendPeriod = newSendPeriod;
+    }
+}
+
+void DefaultCC::init() {
+    _lastRCTime = high_resolution_clock::now();
+    
+    _mss = udt::MAX_PACKET_SIZE_WITH_UDP_HEADER;
+    
+    _slowStartLastAck = _sendCurrSeqNum;
+    _lastDecreaseMaxSeq = SequenceNumber { SequenceNumber::MAX };
+    
+    _congestionWindowSize = 16.0;
+    _packetSendPeriod = 1.0;
```

Comment:
I agree, all members/methods should be private by default unless they have a good reason not to be.
And I'm also usually more in favor of all member are private and getter/setter can be protected/public if needed.
Make it so much easier if for some reason you want to had some debug to see when a member is changed/accessed.